csharp/sample/Microsoft.ML.OnnxRuntime.FasterRcnnSample/README.md

```diff
@@ -0,0 +1,172 @@
+
+```cs
+string modelFilePath = args[0];
+string imageFilePath = args[1];
+string outImageFilePath = args[2];
+```
+
+### Read image
+
+Next, we will read the image in using the cross-platform image library [ImageSharp](https://www.nuget.org/packages/SixLabors.ImageSharp):
+
+```cs
+using Image<Rgb24> image = Image.Load<Rgb24>(imageFilePath, out IImageFormat format);
+```
+
+Note, we're specifically reading the `Rgb24` type so we can efficiently preprocess the image in a later step.
+
+### Resize image
+
+Next, we will resize the image to the appropriate size that the model is expecting; it is recommended to resize the image such that both height and width are within the range of [800, 1333].
+
+```cs
+float ratio = 800f / Math.Min(image.Width, image.Height);
+using Stream imageStream = new MemoryStream();
+image.Mutate(x => x.Resize((int)(ratio * image.Width), (int)(ratio * image.Height)));
+image.Save(imageStream, format);
+```
+
+### Preprocess image
+
+Next, we will preprocess the image according to the [requirements of the model](https://github.com/onnx/models/tree/master/vision/object_detection_segmentation/faster-rcnn#preprocessing-steps):
+
+```cs
+var paddedHeight = (int)(Math.Ceiling(image.Height / 32f) * 32f);
+var paddedWidth = (int)(Math.Ceiling(image.Width / 32f) * 32f);
+Tensor<float> input = new DenseTensor<float>(new[] { 3, paddedHeight, paddedWidth });
+var mean = new[] { 102.9801f, 115.9465f, 122.7717f };
+for (int y = paddedHeight - image.Height; y < image.Height; y++)
+{
+    Span<Rgb24> pixelSpan = image.GetPixelRowSpan(y);
+    for (int x = paddedWidth - image.Width; x < image.Width; x++)
+    {
+        input[0, y, x] = pixelSpan[x].B - mean[0];
+        input[1, y, x] = pixelSpan[x].G - mean[1];
+        input[2, y, x] = pixelSpan[x].R - mean[2];
+    }
+}
+```
+
+Here, we're creating a Tensor of the required size `(channels, paddedHeight, paddedWidth)`, accessing the pixel values, preprocessing them and finally assigning them to the tensor at the appropriate indicies.
+
+### Setup inputs
+
+Next, we will create the inputs to the model:
+
+```cs
+var inputs = new List<NamedOnnxValue>
+{
+    NamedOnnxValue.CreateFromTensor("image", input)
+};
+```
+
+To check the input node names for an ONNX model, you can use [Netron](https://github.com/lutzroeder/netron) to visualise the model and see input/output names. In this case, this model has `image` as the input node name.
+
+### Run inference
+
+Next, we will create an inference session and run the input through it:
+
+```cs
+using var session = new InferenceSession(modelFilePath);
+using IDisposableReadOnlyCollection<DisposableNamedOnnxValue> results = session.Run(inputs);
+```
+
+### Postprocess output
+
+Next, we will need to postprocess the output to get boxes and associated label and confidence scores for each box:
+
+```cs
+var resultsArray = results.ToArray();
+float[] boxes = resultsArray[0].AsEnumerable<float>().ToArray();
```

Comment:
This really should be `.AsEnumerable<float[]>().ToArray();` but I was getting NRE. The above works but is not intuitive, given the boxes are actually output as `float32[nbox,4]`. Let me know if there's a better way to extract the boxes.